### PR TITLE
Fix the build on Windows and add GitHub Action

### PR DESCRIPTION
**Pull request by steveklabnik**
Let's see if I did the action right. 🤞

### CODE REVIEW

**Comment by steveklabnik** [comment]
I do not understand why the "check", "clippy" and "test suite" jobs aren't running. It's late here, and I'm off tomorrow, so I'm going to leave this as-is; if anyone wants to fix the Windows build in the meantime, pulling in the first commit should work just fine.

File: cmd/rpc/Cargo.toml [hunk 1]
```diff
@@ -21,3 +21,6 @@ libc = "0.2"
 log = {version = "0.4.8", features = ["std"]}
 parse_int = "0.4.0"
 zerocopy = "0.6.1"
+
+[target.'cfg(windows)'.dependencies]
```
Comment (mkeeter) [inline cmd/rpc/Cargo.toml:25]: Nit: do we want to make the `libc` dependency only included on `not(windows)`?

Comment (steveklabnik) [reply cmd/rpc/Cargo.toml:25]: Seems fine to me, I was being lazy; it's probably transitively included anyway.

**Comment by steveklabnik** [comment]
Looking again, they still say the status haven't been reported, but if I dig way down:

image

So, an error that doesn't show up as an error. Sigh. At least this points to what needs fixed!

**Comment by steveklabnik** [comment]
Okay this is looking good! Turns out an array of `runs-on` is not the behavior I thought it was, haha. I've enabled auto-merge so that we can get to testing asap; if anyone has any other tweaks they'd like me to make here, please let me know and I'd be happy to make a follow-up PR.

**Comment by steveklabnik** [comment]
I am *very* confused about these top-level "Check" and "Test Suite" jobs that aren't running. The two jobs named these things are both under "Continuous integration" and are shown above, and working. When I go to the "checks" tab, these two things don't show up. wtf.

**Comment by steveklabnik** [comment]
Thank you @alilleybrinker and others; this was in fact a branch permissions thing. Fixed now!

**Pull request merged**